worker/scraper.py

```diff
@@ -49,6 +55,15 @@ def categorize_feed_articles(feed, categorizer):
         for category in categories:
             item.categories.append(category)
 
+            # Mark category as unread for all users
+            try:
+                category_model = Category.query.filter_by(category=category).one()
+                for user_category in UserCategory.query.filter_by(category=category_model).all():
```

Comment:
Just learned about: http://docs.sqlalchemy.org/en/latest/core/dml.html

Want to give that a shot? Looks nifty!